[2025-03-17T02:04:38.314+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T02:04:36.550022+00:00 [queued]>
[2025-03-17T02:04:38.322+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T02:04:36.550022+00:00 [queued]>
[2025-03-17T02:04:38.324+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-03-17T02:04:38.336+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): reddit_extraction> on 2025-03-17 02:04:36.550022+00:00
[2025-03-17T02:04:38.344+0000] {standard_task_runner.py:57} INFO - Started process 51 to run task
[2025-03-17T02:04:38.346+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'etl_reddit_pipeline', 'reddit_extraction', 'manual__2025-03-17T02:04:36.550022+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmp44l46a9d']
[2025-03-17T02:04:38.347+0000] {standard_task_runner.py:85} INFO - Job 3: Subtask reddit_extraction
[2025-03-17T02:04:38.389+0000] {task_command.py:415} INFO - Running <TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T02:04:36.550022+00:00 [running]> on host 1ba35855be5c
[2025-03-17T02:04:38.449+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='vijay Mopuru' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='reddit_extraction' AIRFLOW_CTX_EXECUTION_DATE='2025-03-17T02:04:36.550022+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-17T02:04:36.550022+00:00'
[2025-03-17T02:04:38.639+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/reddit_dag.py", line 11, in reddit_pipeline
    reddit = praw.Reddit(
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/util/deprecate_args.py", line 46, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/reddit.py", line 259, in __init__
    raise MissingRequiredAttributeException(
praw.exceptions.MissingRequiredAttributeException: Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.
[2025-03-17T02:04:38.652+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_reddit_pipeline, task_id=reddit_extraction, execution_date=20250317T020436, start_date=20250317T020438, end_date=20250317T020438
[2025-03-17T02:04:38.665+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 3 for task reddit_extraction (Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.; 51)
[2025-03-17T02:04:38.696+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-03-17T02:04:38.707+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
